app/models/application_record.rb

```diff
@@ -26,6 +26,29 @@ def self.estimated_count
     count
   end
 
+  # Throughout the code base we make use of a lot of polymorphic relations (e.g. acts_as_follower,
+  # acts_as_taggable, etc.).  Which means we often want to know the name of the relationship.
+  #
+  # Furthermore in our views and controllers we've often relied on using `object.class.name` for
+  # branching logic.
+  #
+  # However, with the sometimes transparent usage of the ApplicationDecorator, we can get unexpected
+  # results.
+  #
+  # This method is here to help us move past the somewhat fragile method chain of
+  # `object.class.name`.
+  #
+  # @return [String]
+  #
+  # @see ApplicationDecorator
+  # @see #decorate
+  #
+  # @todo We may want to consider adding a class method that is polymorphic_type_name; but for now
+  #       we'll use an instance variable.
+  def polymorphic_type_name
+    self.class.name
```

Comment:
so right now this _just_ names the concept (and does a more or less uniform replacement of the existing uses of `record.class.name` -> `record.polymorphic_type_name` when used for dispatch)? 

I'm on board, it would be doubly good if it were somehow a shorter string, but you type once and read ten times.

Reply:
I think `record.type_name` would be a reasonable alternative.

Reply:
There's also the `record.model_name`

```ruby
[1] pry(main)> User.model_name
=> #<ActiveModel::Name:0x0000000106c01780
 @collection="users",
 @element="user",
 @human="User",
 @i18n_key=:user,
 @klass=User (call 'User.connection' to establish a connection),
 @name="User",
 @param_key="user",
 @plural="users",
 @route_key="users",
 @singular="user",
 @singular_route_key="user">
[2] pry(main)> User.new.model_name
=> #<ActiveModel::Name:0x0000000106c01780
 @collection="users",
 @element="user",
 @human="User",
 @i18n_key=:user,
 @klass=
  User(id: integer, apple_username: string, articles_count: integer, badge_achievements_count: integer, blocked_by_count: integer, blocking_others_count: integer, checked_code_of_conduct: boolean, checked_terms_and_conditions: boolean, comments_count: integer, confirmation_sent_at: datetime, confirmation_token: string, confirmed_at: datetime, created_at: datetime, credits_count: integer, current_sign_in_at: datetime, current_sign_in_ip: inet, email: string, encrypted_password: string, export_requested: boolean, exported_at: datetime, facebook_username: string, failed_attempts: integer, feed_fetched_at: datetime, following_orgs_count: integer, following_tags_count: integer, following_users_count: integer, forem_username: string, github_repos_updated_at: datetime, github_username: string, invitation_accepted_at: datetime, invitation_created_at: datetime, invitation_limit: integer, invitation_sent_at: datetime, invitation_token: string, invitations_count: integer, invited_by_id: integer, invited_by_type: string, last_article_at: datetime, last_comment_at: datetime, last_followed_at: datetime, last_moderation_notification: datetime, last_notification_activity: datetime, last_onboarding_page: string, last_reacted_at: datetime, last_sign_in_at: datetime, last_sign_in_ip: inet, latest_article_updated_at: datetime, locked_at: datetime, monthly_dues: integer, name: string, old_old_username: string, old_username: string, onboarding_package_requested: boolean, organization_info_updated_at: datetime, payment_pointer: string, profile_image: string, profile_updated_at: datetime, rating_votes_count: integer, reactions_count: integer, registered: boolean, registered_at: datetime, remember_created_at: datetime, remember_token: string, reputation_modifier: float, reset_password_sent_at: datetime, reset_password_token: string, saw_onboarding: boolean, score: integer, secret: string, sign_in_count: integer, signup_cta_variant: string, spent_credits_count: integer, stripe_id_code: string, subscribed_to_user_subscriptions_count: integer, twitter_username: string, unconfirmed_email: string, unlock_token: string, unspent_credits_count: integer, updated_at: datetime, username: string, workshop_expiration: datetime),
 @name="User",
 @param_key="user",
 @plural="users",
 @route_key="users",
 @singular="user",
 @singular_route_key="user">
```

Reply:
`model_name` seems clear to me, but creates an assumption that the "type" we are dispatching on is a model, or that the model is an application record - that might shape uses in the future. I'm not _opposed_ to the long name, I prefer short names when they exist and are still clear.

Reply:
One time our DBA opted for the field name "electronic_financial_transaction_vehicle_type".  The range of values was "Credit Card", "ACH".  I favor longer names earlier because `sed` makes it easier to shorten later.

Reply:
Like @djuber I have some concerns about the name of this method. It's not so much the length of it but that it sounds a bit prescriptive. "Am I not supposed to use this when I need the string representation of a class name in a context that is not setting a polymorphic attribute?" is probably something I'd ask myself if I'm not aware of the background. I would prefer the simple `class_name`, which is exactly as long as the old version, not prescriptive and as far as I'm concerned clear enough.